### PR TITLE
[DC-1664] CT is not repopulating `ethnicity_concept_id` the same as RT

### PR DESCRIPTION
**Pull request by jp3477**
Expands the subgroups used in repopulation ethnicity person fields.

### CODE REVIEW

File: data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier.py [hunk 1]
```diff
@@ -180,6 +190,60 @@ class JoinExpression(NamedTuple):
...
+        END AS {{prefix}}_concept_id
+    )
+    {% endif %}
+FROM 
+(
+    SELECT
+        p.person_id,
+        COALESCE(ethnicity_ob.value_as_concept_id,
+            CASE race_ob.value_source_concept_id
+                WHEN  {{no_matching_concept_id}} THEN {{no_matching_concept_id}} /*missing answer*/
```
Comment (nishanthpp93) [inline data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier.py:213]: Post discussion notes:
1. Update to IF statement to have the same behavior as RT.

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier_test.py [hunk 2]
```diff
@@ -106,20 +107,25 @@ def setUp(self):
...
-                (6, 2, 45880669, 1585838, 1585839, '2020-01-01', 0, 0),
-                (7, 2, 1585841, 1585838, 1585841, '2020-01-01', 0, 0),
-                (8, 1, 45878463, 1585845, 1585847, '2020-01-01', 0, 0),
-                (9, 2, 45880669, 1585845, 1585846, '2020-01-01', 0, 0)
+                (1, 1, 45877987, 1586140, 1586146, 'WhatRaceEthnicity_White', '2020-01-01', 0, 0),
+                (2, 1, 1586143, 1586140, 1586143, 'WhatRaceEthnicity_Black', '2020-01-01', 0, 0),
+                (3, 2, 45879439, 1586140, 1586142, 'WhatRaceEthnicity_Asian', '2020-01-01', 0, 0),
+                (4, 2, 1586147, 1586140, 1586147, 'WhatRaceEthnicity_Hispanic', '2020-01-01', 0, 0),
+                (5, 1, 45878463, 1585838, 1585840, 'GenderIdentity_Woman', '2020-01-01', 0, 0),
+                (6, 2, 45880669, 1585838, 1585839, 'GenderIdentity_Man', '2020-01-01', 0, 0),
```
Comment (nishanthpp93) [inline tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier_test.py:121]: Tests are great! Thanks for adding the `value_source_value`, makes it easier to understand.

**Review by nishanthpp93 — CHANGES_REQUESTED** [review]
Overall does the same as RT except for a small change suggested.

File: data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier.py (hunk 1 above)
Comment (jp3477) [reply data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier.py]: Just added the change.

File: data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier.py [hunk 3]
```diff
@@ -180,6 +190,68 @@ class JoinExpression(NamedTuple):
...
+        p.person_id,
+        IF
+        (ethnicity_ob.value_as_concept_id IS NULL,
+            /*Case this out based on the race_ob (race) values, ie if it's a skip/pna respect that.*/
+            CASE race_ob.value_source_concept_id
+                WHEN  {{no_matching_concept_id}} THEN {{no_matching_concept_id}} /*missing answer*/
+                WHEN  NULL THEN {{no_matching_concept_id}} /*missing answer*/
+                WHEN  {{pna_concept_id}} THEN {{pna_concept_id}} /*PNA*/
+                WHEN  {{skip_concept_id}} THEN {{skip_concept_id}} /*Skip*/
+                WHEN  {{none_of_these_concept_id}} THEN {{none_of_these_concept_id}} /*None of these*/
```
Comment (nishanthpp93) [inline data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier.py:219]: The `NULL` check doesn't work as expected like you mentioned, but can be fixed using:
```suggestion
            CASE 
                WHEN race_ob.value_source_concept_id = {{no_matching_concept_id}} THEN {{no_matching_concept_id}} /*missing answer*/
                WHEN race_ob.value_source_concept_id IS NULL THEN {{no_matching_concept_id}} /*missing answer*/
                WHEN race_ob.value_source_concept_id = {{pna_concept_id}} THEN {{pna_concept_id}} /*PNA*/
                WHEN race_ob.value_source_concept_id = {{skip_concept_id}} THEN {{skip_concept_id}} /*Skip*/
                WHEN race_ob.value_source_concept_id = {{none_of_these_concept_id}} THEN {{none_of_these_concept_id}} /*None of these*/
```

Do you think you can make this specific change in the RT too? We can make it in a separate ticket if you prefer, but we definitely need it before the CDR generation.

**Review by nishanthpp93 — COMMENTED** [review]
Thanks for updating it, added one more suggestion. Let me know your thoughts.

File: data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier.py (hunk 3 above)
Comment (jp3477) [reply data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier.py]: Thanks for the reminder, adding it to RT now.

Comment (jp3477) [reply data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier.py]: Made the change to RT as well. I had to slightly modify the integration test.

**Review by nishanthpp93 — APPROVED** [review]
Thanks for making all the changes and updating the RT and tests, this is good to merge!

**Pull request merged**